test/common/index.js

```diff
@@ -698,19 +698,27 @@ Object.defineProperty(exports, 'hasSmallICU', {
 });
 
 // Useful for testing expected internal/error objects
-exports.expectsError = function expectsError({code, type, message}) {
+exports.expectsError = function expectsError(options) {
   return function(error) {
-    assert.strictEqual(error.code, code);
-    if (type !== undefined) {
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
-    }
-    if (message instanceof RegExp) {
-      assert(message.test(error.message),
-             `${error.message} does not match ${message}`);
-    } else if (typeof message === 'string') {
-      assert.strictEqual(error.message, message);
-    }
+    Object.keys(options).forEach((key) => {
+      if (key === 'type') {
+        const type = options.type;
+        if (type !== undefined) {
+          assert(error instanceof type,
+                 `${error} is not the expected type ${type}`);
+        }
+      } else if (key === 'message') {
+        const message = options.message;
+        if (message instanceof RegExp) {
+          assert(message.test(error.message),
+                 `${error.message} does not match ${message}`);
+        } else if (typeof message === 'string') {
+          assert.strictEqual(error.message, message);
+        }
+      } else {
+        assert.strictEqual(error[key], options[key]);
```

Comment:
I'm thinking this needs a custom message `The ${key} field of ${error} was ${error[key]} and not {options[key]} as expected` (or a little bit shorter)
Otherwise we get an `AssertionError: a !== b` thrown from deep within the harness.
So the same applies for L716 and the `message` strings.